### PR TITLE
Anya/convert list documents to find document series reference

### PR DESCRIPTION
**Pull request by aroltsch**
FindDocumentSeriesReference returns a list of DocumentSeries objects containing the metadata
for documents that match search criteria. It maps to listDocuments in eDocument Service v4 (deprecated)

The API document is attached to #125 

Shane and Alex, please take a look at the PR and we can have a discussion if there is anything we would like to change around the approach I took. Thank you!

connects #125

### CODE REVIEW

File: lib/vbms/helpers/xml_helper.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require 'nori'
+
+module XMLHelper
```
Comment (shanear) [inline lib/vbms/helpers/xml_helper.rb]: Some unit tests for this module might be good

File: lib/vbms/requests/find_document_series_reference.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        doc.xpath(
+          "//efol:findDocumentSeriesReferenceResponse/efol:result", VBMS::XML_NAMESPACES
+        ).map do |el|
+          constuct_response(XMLHelper.convert_to_hash(el.to_xml)[:result])
+        end
+      end
+
+      private
+
+      def constuct_response(result)
```
Comment (amprokop) [inline lib/vbms/requests/find_document_series_reference.rb]: typo? `construct_response`

File: lib/vbms/requests/find_document_series_reference.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        [["/soapenv:Envelope/soapenv:Body",
+          { soapenv: SoapScum::XMLNamespaces::SOAPENV },
+          "Content"]]
+      end
+
+      def handle_response(doc)
+        doc.xpath(
+          "//efol:findDocumentSeriesReferenceResponse/efol:result", VBMS::XML_NAMESPACES
+        ).map do |el|
+          constuct_response(XMLHelper.convert_to_hash(el.to_xml)[:result])
```
Comment (amprokop) [inline lib/vbms/requests/find_document_series_reference.rb]: typo? `construct_response`

File: lib/vbms/requests/find_document_series_reference.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
+module VBMS
+  module Requests
+    # This call returns a list of DocumentSeries objects containing the metadata
+    # for documents that match search criteria.
+    # It maps to listDocuments in eDocument Service v4 (deprecated)
```
Comment (amprokop) [inline lib/vbms/requests/find_document_series_reference.rb]: Is this comment old? or would it be a little clearer to say `# This service replaces listDocuments in eDocument Service v4, which is deprecated as of March 2017`

File: lib/vbms/requests/find_document_series_reference.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
+module VBMS
+  module Requests
+    # This call returns a list of DocumentSeries objects containing the metadata
+    # for documents that match search criteria.
```
Comment (amprokop) [inline lib/vbms/requests/find_document_series_reference.rb]: Is `file_number` the only criteria we currently use to search?

Comment (amprokop) [inline lib/vbms/requests/find_document_series_reference.rb]: i.e., would it be possible using this gem to search only for a Form 9, or would i have to use this service to get all the documents for a veteran and then look for the Form 9?

File: lib/vbms/helpers/xml_helper.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+require 'nori'
```
Comment (amprokop) [inline lib/vbms/helpers/xml_helper.rb]: i'm curious— why do we need this file now when we didn't before? how were we previously converting xml to a hash?

File: lib/vbms/helpers/xml_helper.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+                         parser: :nokogiri,
+                         strip_namespaces: true)
+  end
+
+  # metadata is an array of hashes
+  # example => [ { :value => "Joe", :@key => "VeteranFirstName" },
+  #              { :value => "Snuffy", :@key => "VeteranLastName" },
+  #              { :value => false, :@key => "restricted" } ]
+  def self.extract_value(metadata, key)
+    metadata.select { |i| i[:@key] == key }[0]
```
Comment (amprokop) [inline lib/vbms/helpers/xml_helper.rb]: not familiar with this syntax— what does the `@` in `i[:@key]` do?

File: lib/vbms/helpers/xml_helper.rb (hunk 6 above)
Comment (aroltsch) [inline lib/vbms/helpers/xml_helper.rb]: Perviously, we were not converting it to hash and it was a pain to use xpath to search for elements. Example: `el.at_xpath("ns2:receivedDt/text()", VBMS::XML_NAMESPACES)`

File: lib/vbms/helpers/xml_helper.rb (hunk 7 above)
Comment (aroltsch) [inline lib/vbms/helpers/xml_helper.rb]: When Nori gem converts XML to hash, it converts class attributes to have `@` in front of them so we can easily identify class attributes.

File: lib/vbms/requests/find_document_series_reference.rb (hunk 4 above)
Comment (aroltsch) [inline lib/vbms/requests/find_document_series_reference.rb]: I added it. I like your correction though!

File: lib/vbms/helpers/xml_helper.rb (hunk 1 above)
Comment (aroltsch) [inline lib/vbms/helpers/xml_helper.rb]: Great idea!

**Pull request merged**